src/Cdrv8833.cpp

```diff
@@ -66,29 +68,53 @@ bool Cdrv8833::move(int8_t power) {
 	if (power > 0) { // forward
 		if (drv8833DecayFast == m_decayMode) {
 			// forward fast decay
-			ledcDetachPin(m_in2Pin);
-			digitalWrite(m_in2Pin, LOW);
-			ledcAttachPin(m_in1Pin, m_channel);
+			#if ESP_IDF_VERSION_MAJOR < 5 
+				ledcDetachPin(m_in2Pin);
+				digitalWrite(m_in2Pin, LOW);
+				ledcAttachPin(m_in1Pin, m_channel);
+			#else
+				ledcDetach(m_in2Pin);
+				digitalWrite(m_in2Pin, LOW);
+				ledcAttachChannel(m_in1Pin, PWM_FREQUENCY, PWM_BIT_RESOLUTION, m_channel);			    
+			#endif
 		}
 		else {
 			// forward slow decay
-			ledcDetachPin(m_in1Pin);
-			digitalWrite(m_in1Pin, HIGH);
-			ledcAttachPin(m_in2Pin, m_channel);
+			#if ESP_IDF_VERSION_MAJOR < 5 
+				ledcDetachPin(m_in1Pin);
+				digitalWrite(m_in1Pin, HIGH);
+				ledcAttachPin(m_in2Pin, m_channel);
+			#else
+				ledcDetach(m_in1Pin);
+				digitalWrite(m_in1Pin, LOW);
+				ledcAttachChannel(m_in2Pin, PWM_FREQUENCY, PWM_BIT_RESOLUTION, m_channel);			    
+			#endif
 		}
 	}
 	else { // reverse
 		if (drv8833DecayFast == m_decayMode) {
 			// reverse fast decay
-			ledcDetachPin(m_in1Pin);
-			digitalWrite(m_in1Pin, LOW);
-			ledcAttachPin(m_in2Pin, m_channel);
+			#if ESP_IDF_VERSION_MAJOR < 5 
+				ledcDetachPin(m_in1Pin);
+				digitalWrite(m_in1Pin, LOW);
+				ledcAttachPin(m_in2Pin, m_channel);
+			#else
+				ledcDetach(m_in1Pin);
+				digitalWrite(m_in1Pin, LOW);
+				ledcAttachChannel(m_in2Pin, PWM_FREQUENCY, PWM_BIT_RESOLUTION, m_channel);			
+			#endif
 		}
 		else {
 			// reverse slow decay
-			ledcDetachPin(m_in2Pin);
-			digitalWrite(m_in2Pin, HIGH);
-			ledcAttachPin(m_in1Pin, m_channel);
+			#if ESP_IDF_VERSION_MAJOR < 5 
+				ledcDetachPin(m_in2Pin);
+				digitalWrite(m_in2Pin, HIGH);
+				ledcAttachPin(m_in1Pin, m_channel);
+			#else
+				ledcDetach(m_in2Pin);
+				digitalWrite(m_in2Pin, LOW);
+				ledcAttachChannel(m_in1Pin, PWM_FREQUENCY, PWM_BIT_RESOLUTION, m_channel);			
+			#endif
 		}
 	}
 	ledcWrite(m_channel, dutyCycle);
```

Comment:
This needs to be `ledcWriteChannel` - https://docs.espressif.com/projects/arduino-esp32/en/latest/api/ledc.html#ledcwritechannel